### DISCUSSION TITLE
Is offline support still considered experimental?

### DISCUSSION THREAD

**Discussion by sighmun** [urql-graphql/urql#2685] [Q&A] [answered]
I'm really enjoying using `urql` instead of Apollo (thanks for a great lib!), and I'm keen to start using the offline support in Graphcache for a production app, but I notice the [API docs page for `storage`](https://formidable.com/open-source/urql/docs/api/graphcache/#storage-option) mentions that it's considered to be experimental, however the [main docs page for Offline support](https://formidable.com/open-source/urql/docs/graphcache/offline/) makes no mention of it.

Is the experimental alert in the storage docs still accurate?

**Comment by kitten** [comment, marked as answer]
It's marked as experimental in that select place because we never reached full confidence in it. We know of several projects using Graphcache's offline mode in production, but we were never able to fully test and validate some of our assumptions ourselves. Furthermore, there's still some edge cases we'll need to sort out in terms of its API design.

That's why we don't have 100% confidence in it. So, while you can likely use it as intended and it'll behave as designed, there may be API design edge cases in its usage that we're not completely aware of yet.

**Reply by sighmun** [reply]
Thanks @kitten, that's helpful to know 👍

**Comment by sarink** [comment]
Just to add my experience: We're using graphcache offline in production, the biggest limitation we've come across is that you cannot perform an update to an object that was created while offline. More in point 1 here: https://github.com/FormidableLabs/urql/issues/2586

**Reply by sighmun** [reply]
Thanks @sarink, good to know 👍
